components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/SessionContextCache.java

```diff
@@ -138,6 +150,17 @@ private SessionContextCacheEntry getSessionFromDB(SessionContextCacheKey key) {
 
         if (sessionContextDO != null) {
             cacheEntry = new SessionContextCacheEntry(sessionContextDO);
+            if (cacheEntry.getContext() == null && cacheEntry.getOptimizedSessionContext() != null) {
+                try {
+                    cacheEntry = SessionContextLoader.getInstance().loadSessionContextCacheEntry(cacheEntry);
+                } catch (SessionDataStorageOptimizationException e) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Error occurred while loading the session context with " +
+                                "context id: %s", cacheEntry.getContextIdentifier()));
+                    }
+                    return null;
```

Comment:
Is this a breaking flow condition, if yes add an exception.

Reply:
It would not break the flow and null is handled at places where it is called.